server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -178,6 +180,8 @@ synchronized void releaseCommit(final IndexCommit snapshotCommit) {
         if (refCount == 0) {
             snapshottedCommits.remove(releasingCommit);
         }
+        // The commit can be clean up only if no pending snapshot and it is not the last commit.
+        return refCount == 0 && releasingCommit.equals(lastCommit) == false;
```

Comment:
don't we need a similar check for the safe commit?

Reply:
Yes, it should return false if the releasing snapshot is the safe commit.